pyacq/core/manager.py

```diff
@@ -47,188 +71,127 @@ class Manager(RPCServer):
     addr : str
         The address for the manager's RPC server.
     """
-    
-    # Classes used internally for bookkeeping
-    class _Host(object):
-        def __init__(self, name, addr):
-            self.rpc_address = addr
-            self.rpc_name = name
-            self.client = RPCClient(name, addr)
-            self.nodegroups = {}
-            self.rpc_hostname = addr.partition('//')[2].rpartition(':')[0]
-
-        def add_nodegroup(self, ng):
-            self.nodegroups[ng.rpc_name] = ng
-        
-        def list_nodegroups(self):
-            return list(self.nodegroups.keys())
-
-
-    class _NodeGroup(object):
-        def __init__(self, host, name, addr):
-            self.host = host
-            self.rpc_address = addr
-            self.rpc_name = name
-            self.client = RPCClient(name, addr)
-            self.nodes = {}
-
-        def add_node(self, name, node):
-            self.nodes[name] = node
-
-        def list_nodes(self):
-            return list(self.nodes.keys())
-
-        def delete_node(self, name):
-            del self.nodes[name]
+    def __init__(self):
+        logger.info('Creating new Manager..')
+        self.hosts = {}  # addr:Host
+        self.nodegroups = {}  # name:Nodegroup
         
-            
-    class _Node(object):
-        def __init__(self, nodegroup, name, classname):
-            self.nodegroup = nodegroup
-            self.name = name
-            self.classname = classname
-            self.outputs = []  # list of StreamDef
-    
-    
-    def __init__(self, name, addr, manager_process=None):
-        RPCServer.__init__(self, name, addr)
-        
-        self.hosts = {}  # name:HostProxy
-        self.nodegroups = {}  # name:NodegroupProxy
-        self.nodes = {}  # name:NodeProxy
-        
-        # auto-generated host on the local machine
-        self._default_host = None
+        # Host used for starting nodegroups on the local machine
+        self.default_host = Host('default_host')
```

Comment:
So default Hiost and manager live the same process, this is much simpler. Thanks.